portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,18 +24,24 @@
 /* Servlet that returns some example content.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-   
+  private ArrayList<String> shows = new ArrayList<String>();
```

Comment:
so HttpServlets are singletons (i.e. there's only 1 per App Server).  There are many HTTP requests and each one corresponds to a thread.  So this is a case of shared memory in a multithreaded environment.  You'll need a threadsafe collection.  A one you can use just for the purposes of this starter project is: CopyOnWriteArrayList which is okay for small n.  For large n it isn't recommended.  (read the java docs for it.)

Reply:
Fixed.